### PR TITLE
Zadanie2

### PR DESCRIPTION
**Pull request by Karolina4033**

### CODE REVIEW

**Review by RakPiotr — APPROVED** [review]
✅brak wycieków danych
✅dwa wytrenowane modele z przetestowanymi hiperparametrami
ładnie opisany proces myślowy, i wizualizacje, nie mam żadnych zastrzeżeń.

**Comment by francesspot** [comment]
ładne opisy (choc mogły byc bardziej rozbudowane) i spełnione wymagania zadania

**Review by francesspot — APPROVED** [review]
ładne opisy (choc mogły byc bardziej rozbudowane) i spełnione wymagania zadania

**Review by Karol-Rejek — COMMENTED** [review]
Fajne opisy wyrażające proces myślowy, moim zdaniem dobrze wykonana praca.

**Comment by Marcelele-0** [comment]
Plusy
- Unikanie data leakage zrobione wręcz książkowo – imputacja braków i skalowanie danych (`StandardScaler`) wykonane poprawnie, dopiero po podziale na train/test.
- Czytelne i bardzo przydatne heatmapy po GridSearchu – widać, że wnioski opierają się na twardych danych z wizualizacji.
- Sprawdzanie metryk na zbiorze treningowym i testowym, żeby na bieżąco kontrolować overfitting. (tutaj uwaga zeby nie tunowac pod test set!)
- Sensownie dobrana siatka parametrów i widać, że autor rozumie, po co skaluje dane przed użyciem KNN.

Minusy
- Zostawione martwe linie w kodzie przy imporcie Dummy (`load_iris` i nadpisanie `X`, `y`), chyba że czegoś nie widzę.
- Optymalizacja w GridSearch ustawiona sztywno pod accuracy – przydałoby się użyć f1-score lub połączyć to z Precision/Recall, o których autor sam pisze (jeśli to przemyślana decyzja, warto wyjaśnić dlaczego).

Ogólnie: Solidny pipeline z dobrą praktyką unikania wycieku danych i fajnymi wizualizacjami. Trzeba tylko wyczyścić kod.

**Review by Marcelele-0 — APPROVED** [review]
